modules/charon-core/src/main/java/org/wso2/charon3/core/schema/AttributeSchema.java

```diff
@@ -71,5 +72,15 @@
 
     public void removeSubAttribute(String subAttributeName) throws CharonException;
 
+    /**
+     * @return Get list of canonical values.
+     */
+    List<String> getCanonicalValues();
+
+    /**
+     * @param canonicalValues canonical values to set.
```

Comment:
Method descriptions are missing? Shall we add it?

Reply:
addressed